Android.md

```diff
@@ -0,0 +1,216 @@
+# Android
+
+## Code
+
+**Code should match JAVA standards. See [relative guide](Java.md) for more details.**
+
+### Architecture
+
+You should separate activities from adapters. Ditto for fragments. 
+
+```
+views/
+  |-- activities/
+  |-- fragments/     // Gathers dialogs and fragments
+  |-- adapters/
+```
+
+### Performances
+
+#### 65k limit
+
+Android apps must have less than 65K methods, including libraries. Then, you should be attentive to import only methods/classes you need (even if there is a work around to overcome this problem).
+
+Remove useless code. Proguard can help you for that.
+
+#### Avoids getters & setters
+
+Even if it is usually a good pattern to follow, you should avoid having getters and setters to access fields in Android. Direct access can improve performances from 3x to 7x. 
+
+#### Do not block UI thread (aka main thread)
+
+Anytime, you should steer clear of blocking UI thread. Indeed, when you are performing long operations, such as pulling data from remote server or engining random stuff in business part, you are freezing your app. User cannot perform interactions anymore.
+
+For instance, avoid to set adapters, pull data from database or server on main thread. To get round this problem, use threads. In particular, Android provides to developers the `AsyncTask` interface. 
+
+#### Avoid Java hashes
+
+Android development implies to be careful about performances and memory management. Instead of using Java hashes (such as `HashMap`), you should use built-in ones, as `SparseArray`. They are designed for being reverent of Android devices. 
+
+#### Manages your memory
+
+Memory is an important resource and quite limited. You should be attentive to free objets, resources and assets as soon as possible. Stop operations when your activity is hidden. Do not let services running for peccadilloes.
+
+#### Do not use enums
+
+Enums are almost 3x eager in memory than static fields. You should use those ones instead, even if for integers.
+
+#### Use static methods (if possible)
+
+Static calls may be 3x faster than classic ones. Use them as much as possible, especially when you need no fields or context (e.g. helpers).
+
+#### Use `float` instead of `double`
+
+Double uses 2x more space than float. What are you still using them?
```

Comment:
I remember hearing this advice back when everyone was using 32-bit processors, but with 64-bit processors, is it really an issue still?